lib/node/runner/checker.go

```diff
@@ -95,10 +95,45 @@ func BallotNotFromKnownValidators(c common.Checker, args ...interface{}) (err er
 	return
 }
 
-// BallotSaveNodeHeight saves height of ballot sender in ISAAC.
-func BallotSaveNodeHeight(c common.Checker, args ...interface{}) (err error) {
+// BallotCheckSYNC performs sync by considering sync condition.
+// And to participate in the consensus,
+// update the latestblock by referring to the database.
+func BallotCheckSYNC(c common.Checker, args ...interface{}) (err error) {
 	checker := c.(*BallotChecker)
-	checker.NodeRunner.Consensus().SaveNodeHeight(checker.Ballot)
+
+	b := checker.Ballot
+	is := checker.NodeRunner.Consensus()
+	is.SaveNodeHeight(b.Source(), b.Round().BlockHeight)
+
+	var height uint64
+	var nodeAddrs []string
+	height, nodeAddrs, err = checker.NodeRunner.Consensus().GetSyncInfo()
+	if err != nil {
+		return err
+	}
+
+	checker.NodeRunner.log.Debug("check GetSyncInfo()", "len(nodeAddrs)", len(nodeAddrs))
+	if is.LatestConfirmedBlock().Height < height {
+		is.StartSync(height, nodeAddrs)
+
+		var blk block.Block
+		blk, err = block.GetLatestBlock(checker.NodeRunner.storage)
+		checker.NodeRunner.Log().Debug("NodeState is SYNC and GetLatestBlock", "height", blk.Height, "err", err)
+		if err != nil {
+			return
+		}
+
+		if is.LatestConfirmedBlock().Height < blk.Height {
```

Comment:
`is.LatestConfirmedBlock()` must be same with `block.GetLatestBlock()`, right? If true, comparing height is already done In https://github.com/bosnet/sebak/pull/524/files#diff-7f79ac520823e445bff5cd70f53bcf59R116 , so how this code have some meaning?

Reply:
@spikeekips 
The `height` in https://github.com/bosnet/sebak/pull/524/files#diff-7f79ac520823e445bff5cd70f53bcf59R116 is the height of the other nodes(from received ballots). But `blk.Height` is latestHeight itself.
The node should check after sync and set latestConsensusedBlock in https://github.com/bosnet/sebak/pull/524/files#diff-7f79ac520823e445bff5cd70f53bcf59R128 for returning to consensus process.